elastic/kibana
# [Response Ops][Alerting] Reusable functions for FAAD resource installation

Resolves https://github.com/elastic/kibana/issues/152490

## Summary

This PR refactors the resource installation methods in `AlertsService` to be reusable library functions. It exports them from the alerting plugin and changes the rule registry resource installer to use them as well.

## To Verify
1. Run this branch with `enableFrameworkAlerts: true`. Verify that we can create a detection rule in the default space & a different space and generate a rule preview. The logs should show that the rule registry creates the resources for the preview indices and for indices in the non-default space.
2. Verify that when running this branch on rule registry rules from `main` or a previous version, the rules continue to run successfully.
